common/containers/Tabs/ENS/components/NameInput/components/HOC.jsx

```diff
@@ -0,0 +1,45 @@
+// @flow
+import * as React from 'react';
+import { isValidENSName } from 'libs/validators';
+type State = {
+  domainToCheck: string,
+  isValidDomain: boolean
+};
+type Props = {
+  resolveDomainRequested: (domain: string) => void
+};
+type HOCPassedComponent = any; //TODO: ComponentType (Property not found in exports of "react")
+
+const HOC = (PassedComponent: HOCPassedComponent) =>
```

Comment:
A more descriptive name of what the HOC does would be nice.

Reply:
(And make searching for the right HOC less collisiony, if we keep this pattern up.)

Reply:
`ENSNameSearchHoc`?